src/nl/Stratego/Bord.java

```diff
@@ -78,6 +78,24 @@ public boolean pieceCheck (int pionYLocation, int pionXLocation, Speler spelerAa
         }
     }
 
+    boolean checkValidPiece(int pionYlocation, int pionXLocation, int team){
+        Object gekozenStuk = speelBord[pionYlocation][pionXLocation];
+        if(gekozenStuk == blokkade){
+            System.out.println("Je hebt een blokkade gekozen");
+            return false;
+        } else if (gekozenStuk == null){
+            System.out.println("Je hebt een lege plek gekozen");
+            return false;
+        } else {
+            Speelstuk gekozenSpeelStuk = (Speelstuk)gekozenStuk; //casten naar een Speelstuk object
+            if (gekozenSpeelStuk.getTeam() != team){
```

Comment:
if (   ((speelstuk)gekozenStuk).getTeam() != team){...}
Dit had ik toen in de chat aan raymond gevraagd en dit zou moeten werken. Weer een variabele minder voor de G.C. om op te ruimen :p